strings/naive_pattern_matching.cpp

```diff
@@ -0,0 +1,42 @@
+#include <iostream>
```

Comment:
https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md
please read this and make changes accordingly! needs documentation and proper tests